Advanced example

bwockner opened the pull request:
Added advanced examples scene and associated assets,
Updated the vdkCollider to lock correctly to a target object; 
Updated documentation

Commit:
Added basic sample and documentation for Unity

Commit:
Re-added meta files to fix unity import issue

Commit:
Merge remote-tracking branch 'origin/master' into bwockner/unitysample

Commit:
Added Ignore

Commit:
Fix crash when attempting to load null UDS

Commit:
Fixed formatting of cs files and changed project files

Also reversed order of try reconnect and connect in login to speed up login process

Commit:
Make Collider Rotate with follow object

Currently causes issues with multiple colliders on one object as rotation is then locked to rotation of follow object and offsets cannot be set by changing the transform of the object

Commit:
Formatting fixes

Commit:
Documentation and formatting fix

Commit:
Merge branch 'bwockner/unitysample' into rotateColliders

Commit:
Added Renew license to API, updated doc

Commit:
Changed more whitespace

Commit:
Updated doc formatting

Commit:
Started Advanced Integration

Commit:
Added option to disable collider rotation with target

Commit:
Added option to disable collider rotation with target

Commit:
Fixed no rotate option on collider, started Advanced Scene

Advanced scene working with ragdoll model

Commit:
Fixed collider rotation bug

Commit:
Code Formatting and cleanup (no functional changes)

Commit:
Readme typo fixes

Commit:
More formatting fixes

Commit:
fix rotation to body

Commit:
API Fix

Commit:
Merge branch 'bwockner/unitysample' into AdvancedExample

Commit:
Added Vehicle examples and simplified collider

Commit:
Cleaned up collider logic, changed rotation locking to be by individual axis

Commit:
Updated Advanced scene

Commit:
Update Docs with Advanced Examples

Commit:
Added renew license thread, imporved collider rotate

Commit:
Finalized rotation with body (now locks to object with offset). Formatting Changes. Fixed load all UDS in directory

Commit:
Moved third party assets to separate folder

Commit:
Updated README.md

Commit:
Merge remote-tracking branch 'remotes/origin/master' into AdvancedExample

Commit:
re-moved files added by previous merge

Commit:
Updated docs with images of advanced samples

OsiJr committed:
Added Convert Custom sample

Commit:
Updated c# API to include picking and options. Added lines to c# example to test picking, currently does not work

Commit:
Implemented Picking interface

Commit:
Added picking and raycasting examples to basic example

Commit:
Added missing .cs files

Commit:
commented out expensive call in PPES, added support for scene camera rendering options

Commit:
commented out expensive call in PPES, added support for scene camera rendering options

Commit:
Merge branch 'AdvancedExample' of github.com:Euclideon/vaultsdksamples into AdvancedExample

OsiJr committed:
Added Esri SceneView integration sample

Commit:
Fixed missing bool

- Added an expected bool

Commit:
Merge branch 'AdvancedExample' of github.com:Euclideon/vaultsdksamples into AdvancedExample

Commit:
Added required tag

- Tag missing caused null reference, added in

Commit:
Updated Unity Package

- Updated README.md to better reflect the current state of installing the VDK with Unity
- Removed vaultSDK.dll meta data
- Junk meta data for vaultSDK.DLL and vaultSDK.lib accidentally made their way into a push and removed
- Added VDK toolbar menu so the user can save their login details in editor
- Removed scene camera tag throwing errors in blank projects
- Added directional light to sample scene
- Modified the readme file to remove old walk through regarding solving errors associated with post processing package.
- Included package manifest.json
- Included package manifest json file to remove null refrences/errors associated with the post processing package. This manifest file also includes a load of extra packages and ideally the default project would have as little package requirements as possible.
- Tag missing caused null reference, added in

Commit:
Included package manifest.json

- Included package manifest json file to remove null refrences/errors associated with the post processing package. This manifest file also includes a load of extra packages and ideally the default project would have as little package requirements as possible.

Commit:
Changed README to reflect packages

- Modified the readme file to remove old walk through regarding solving errors associated with post processing package.

Commit:
Adding VDK toolbar menu

- Added VDK toolbar menu so the user can save their login details in editor
- Removed scene camera tag throwing errors in blank projects
- Added directional light to sample scene

Commit:
Removed vaultSDK.dll meta data

- Junk meta data for vaultSDK.DLL and vaultSDK.lib accidentally made their way into a push and removed

Commit:
-Fixes for cameraOptions and picking system
-Fixed coordinate system transform from UD to OpenGL coordinates
-Fixed multimodel to correctly locate models relative to each other
-Added bool to position model to baseoffset in UDS header, added additional parameter to offset this in cases that location is too large

Commit:
Updated README

- Updated README.md to better reflect the current state of installing the VDK with Unity

Commit:
Merge branch 'AdvancedExample' of github.com:Euclideon/vaultsdksamples into bwockner/geolocationFixes

Commit:
Merge branch 'AdvancedExample' of github.com:Euclideon/vaultsdksamples into bwockner/geolocationFixes

Commit:
Merge branch 'bwockner/geolocationFixes' of github.com:Euclideon/vaultsdksamples into bwockner/geolocationFixes

Commit:
Merge branch 'bwockner/geolocationFixes' into AdvancedExample

Commit:
Modified README

- Modified readme to match the previous MASTEER readme, with only changes relevant to the new login method / setup methods.

OsiJr committed:
Merge pull request #9 from Euclideon/enoble/UnityUX

Enoble/unity ux

Commit:
Update to include function call for convert

Commit:
Merge branch 'master' of https://github.com/Euclideon/vaultsdksamples into ucpasas/pythonwrapperconvertupdate

Commit:
Added simple converter for py 3

OsiJr commented on integrations/unity-csharp/Assets/VDK/UDUtilities.cs:
```diff
@@ -2,12 +2,29 @@
     public static class UDUtilities
     {
+        public static Matrix4x4 UDtoGL =
+                        Matrix4x4.Scale(new Vector3(1, -1, 1)) *
+                        Matrix4x4.Rotate(Quaternion.Euler(90, 0, 0));
```
Indentation

OsiJr commented on integrations/unity-csharp/Assets/VDK/UDUtilities.cs:
```diff
@@ -46,21 +63,90 @@ public static vdkRenderInstance[] getUDSInstances()
+                    Transform localTransform = objects[i].transform;
+
+                    modelArray[count].worldMatrix = UDUtilities.GetUDMatrix(
+                            Matrix4x4.TRS(model.transform.position, model.transform.rotation, model.transform.localScale) *
+                            model.modelToPivot
+                        );
```
Indentation and strangely multiline?

bwockner replied:
Not a fan of cramming complex expressions into one line seeing as whitespace doesn't cost anything... Are we not allowed multiline expressions?

OsiJr commented on integrations/unity-csharp/Assets/Advanced Assets/demoSwitcher.cs:
```diff
@@ -0,0 +1,38 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class demoSwitcher : MonoBehaviour
```
Classes should be ProperCase

OsiJr commented on integrations/unity-csharp/Assets/Advanced Assets/loadAllUDSInDirectory.cs:
```diff
@@ -1,52 +1,76 @@
 using Vault;
 
+/*
+ *Loads all uds files in a local path and places them in the world space relative to the first
```
```suggestion
 * Loads all UDS files in a local path and places them in the world space relative to the first
```

OsiJr commented on integrations/unity-csharp/Assets/Advanced Assets/loadAllUDSInDirectory.cs:
```diff
@@ -1,52 +1,76 @@
+                continue;
+            }
+            if (i == baseInd)
+            {//reference all models to the first 
```
```suggestion
            if (i == baseInd) //reference all models to the first 
            {
```

OsiJr commented on integrations/unity-csharp/Assets/Advanced Assets/loadAllUDSInDirectory.cs:
```diff
@@ -1,52 +1,76 @@
+        if (this.reload)
+        {
+            foreach (Transform child in transform) {
+                GameObject.Destroy(child.gameObject);
```
Is it save to `Destroy` in a `foreach`?

OsiJr commented on integrations/unity-csharp/Assets/VDK/VDKUserInfoEditor.cs:
```diff
@@ -0,0 +1,143 @@
+{
+
+
+
```
So much excessive whitespace

OsiJr commented on integrations/unity-csharp/Assets/vdkCameraOptions.cs:
```diff
@@ -0,0 +1,111 @@
+            cam = Camera.main;
+        }
+    }
+    void Start()
```
```suggestion
    }
    
    void Start()
```

OsiJr commented on integrations/unity-csharp/Assets/VDK/UDSModel.cs:
```diff
@@ -39,7 +38,31 @@ public string Path
+    private void Update()
+    {
+
+        if (geolocate)
+        {
+            this.transform.localPosition =  UDUtilities.UDtoGL *
+                new Vector3(
+                    (float)(header.baseOffset[0] + header.pivot[0] * header.scaledRange),
+                    (float)(header.baseOffset[1] + header.pivot[1] * header.scaledRange),
+                    (float)(header.baseOffset[2] + header.pivot[2] * header.scaledRange)
+                    );
+            transform.localPosition += geolocationOffset;
+            geolocate = false;
+        }
+    }
+    public Matrix4x4 getStoredMatrix() {
+        return new Matrix4x4(
+                new Vector4((float)header.storedMatrix[0], (float)header.storedMatrix[1], (float)header.storedMatrix[2], (float)header.storedMatrix[3]),
+                new Vector4((float)header.storedMatrix[4], (float)header.storedMatrix[5], (float)header.storedMatrix[6], (float)header.storedMatrix[7]),
+                new Vector4((float)header.storedMatrix[8], (float)header.storedMatrix[9], (float)header.storedMatrix[10], (float)header.storedMatrix[11]),
+                new Vector4((float)header.storedMatrix[12], (float)header.storedMatrix[13], (float)header.storedMatrix[14], (float)header.storedMatrix[15])
+                ) ;
```
Indentation and splitting decisions are strange here.

bwockner replied:
Indentation can be fixed (I assume you want the closing paren in line with the opening expressions rather than the contents and contents indented), unsure what to do regarding the splitting, I really don't want either of these expressions to be crammed on the one line as it makes the code extremely unreadable, declaring temporary variables doesn't improve it either IMO.

OsiJr commented on languages/csharp/VaultAPI.cs:
```diff
@@ -3,416 +3,600 @@
+            if (error != vdkError.vE_Success)
+            {
+                throw new Exception("Unable to keep session alive: " + error.ToString());
+            }
```
```suggestion
                throw new Exception("Unable to keep session alive: " + error.ToString());
```

OsiJr commented on languages/csharp/VaultAPI.cs:
```diff
@@ -3,416 +3,600 @@
+
+            vdkError error = vdkContext_TryResume(ref pContext, pURL, pApplicationName, pUsername, tryDongle);
+            if (error != vdkError.vE_Success)
+            {
+                throw new Exception("Unable to keep session alive: " + error.ToString());
+            }
```
```suggestion
                throw new Exception("Unable to keep session alive: " + error.ToString());
```

OsiJr commented on languages/csharp/VaultAPI.cs:
```diff
@@ -3,416 +3,600 @@
+            
+        }
+
+        public void setPick(uint x, uint y)
```
ProperCase for functions

```suggestion
        public void SetPick(uint x, uint y)
```

OsiJr reviewed (changes requested):
I didn't comment everywhere but there is massive inconsistency in;
- tabs vs spaces (either is fine but only 1 type per file...)
- multiline comments for a single line
- camelCase classes or functions (both should be ProperCase)

Also missing is the license information for the third party files
- This repo is MIT but lots of that code and assets likely isn't MIT compatible
- The models don't have sources listed (if they are public domain)

OsiJr committed:
Merge pull request #10 from Euclideon/ucpasas/pythonwrapperconvertupdate

Update to include function call for convert

OsiJr committed:
Minor fixes to convertcustom feature sample

samsinsane committed:
Merge pull request #6 from Euclideon/pfox/convertcustom

Added Convert Custom sample

OsiJr committed:
Added feature sample for rendering using custom voxel shaders

bwockner committed:
Update languages/csharp/VaultAPI.cs

Co-Authored-By: Paul Fox <pfox@euclideon.com>

bwockner committed:
Update integrations/unity-csharp/Assets/vdkCameraOptions.cs

Co-Authored-By: Paul Fox <pfox@euclideon.com>

bwockner committed:
Update integrations/unity-csharp/Assets/Advanced Assets/loadAllUDSInDirectory.cs

Co-Authored-By: Paul Fox <pfox@euclideon.com>

bwockner committed:
Update integrations/unity-csharp/Assets/Advanced Assets/loadAllUDSInDirectory.cs

Co-Authored-By: Paul Fox <pfox@euclideon.com>

OsiJr committed:
Added basic Cesium skeleton

There are 2 major bugs remaining
- Depth isn't being correctly calculated
- There is an ECEF transform bug in Clang in udCore

OsiJr committed:
Fixed copy/paste title error

asunawu committed:
Try to use clipPosition.w as depth.

OsiJr committed:
Merge pull request #13 from Euclideon/awu/FixedCesiumDepth

Try to use clipPosition.w as depth.

OsiJr committed:
Merge remote-tracking branch 'remotes/origin/pfox/esriDemo'

OsiJr committed:
Merge remote-tracking branch 'remotes/origin/pfox/cesium'

samsinsane committed:
Fixed features sample locations for macOS builds

samsinsane committed:
Merge pull request #11 from Euclideon/pfox/voxelShaderSample

Added feature sample for rendering using custom voxel shaders

OsiJr committed:
[Cesium] Converted tabs to spaces and removed trailing white space

OsiJr committed:
[Cesium] Added information about how this integration was created

samsinsane committed:
Merge pull request #15 from Euclideon/pfox/cesiumtidy

[Cesium] Converted tabs to spaces and removed trailing white space

OsiJr committed:
[Cesium] Fixed formatting to use 3 space+1. format

samsinsane committed:
Merge pull request #14 from Euclideon/pfox/cesiumInfo

[Cesium] Added information about how this integration was created

Commit:
Updated to work with VDK0.6

Commit:
Merge branch 'AdvancedExample' of github.com:Euclideon/vaultsdksamples into AdvancedExample

Commit:
Merge branch 'AdvancedExample' of https://github.com/Euclideon/vaultsdksamples

# Conflicts:
#	integrations/unity-csharp/Assets/Advanced Assets/Third Party/Driving/Models/Vehicles/SM_Railjet_FirstCarriage.mtl.meta
#	integrations/unity-csharp/Assets/VDK/VDKRenderProfile.asset
#	integrations/unity-csharp/Packages/manifest.json
#	integrations/unity-csharp/README.md

Commit:
Merge branch 'AdvancedExample' of https://github.com/Euclideon/vaultsdksamples

# Conflicts:
#	integrations/unity-csharp/Assets/Advanced Assets/Third Party/Driving/Models/Vehicles/SM_Railjet_FirstCarriage.mtl.meta
#	integrations/unity-csharp/Assets/VDK/VDKRenderProfile.asset
#	integrations/unity-csharp/Packages/manifest.json
#	integrations/unity-csharp/README.md

Commit:
Merge branch 'AdvancedExample' of https://github.com/Euclideon/vaultsdksamples into unity_login

Commit:
Fixed another merge conflict
